mobile/components/StarRating.tsx

```diff
@@ -0,0 +1,67 @@
+import { Star } from "lucide-react-native";
+import { useEffect, useState } from "react";
+import { TouchableOpacity, Text, View, StyleSheet } from "react-native";
+
+export default function StarRating({ onRate, value = 0 }: { onRate: (rating: number) => void; value?: number }) {
+	const [rating, setRating] = useState(value);
+
+	useEffect(() => {
+		setRating(value);
+	}, [value]);
+
+	const handleStarClick = (starIndex: number) => {
+		const newRating = rating === starIndex ? 0 : starIndex;
+		setRating(newRating);
+		onRate(newRating);
+	};
+
+	return (
+		<View style={styles.container}>
+			<View style={[styles.row, { justifyContent: "center" }]}>
+				<View style={styles.starRow}>
+					{[1, 2, 3, 4, 5].map((starIndex) => (
+						<TouchableOpacity
+							key={starIndex}
+							onPress={() => handleStarClick(starIndex)}
+							accessibilityRole="radio"
+							accessibilityState={{ selected: rating === starIndex }}
+							accessibilityLabel={`Rate ${starIndex} star${starIndex !== 1 ? "s" : ""}`}
+						>
+							<Star
+								size={32}
+								color={starIndex <= rating ? "#FACC15" : "#D1D5DB"}
+								fill={starIndex <= rating ? "#FACC15" : "none"}
+							/>
+						</TouchableOpacity>
+					))}
+				</View>
+			</View>
+		</View>
+	);
+}
+
+const styles = StyleSheet.create({
```

Comment:
I feel like there's a lot of padding on the interface as a whole... The buttons end up way too spaced out and the X button to close the popup ends up almost outside the screen / safe area inset. I believe we probably tested on different devices but I still think reducing paddings throughout might be a good idea
<img width="547" alt="Screenshot 2025-06-28 at 6 34 29 PM" src="https://github.com/user-attachments/assets/0bab9574-43ab-4fa1-9f82-8f67b7a7b277" />